evm/src/executor/inspector/cheatcodes/mod.rs

```diff
@@ -126,7 +126,7 @@ pub struct Cheatcodes {
     pub mocked_calls: BTreeMap<Address, BTreeMap<MockCallDataContext, MockCallReturnData>>,
 
     /// Expected calls
-    pub expected_calls: BTreeMap<Address, Vec<(ExpectedCallData, u64)>>,
+    pub expected_calls: BTreeMap<Address, BTreeMap<Bytes, (ExpectedCallData, u64)>>,
```

Comment:
This is now storing this information

```
BTreeMap<Address, // -> Target
    BTreeMap<Bytes, // -> Calldata. Selectors are NOT deduped, so partial matches and full matches can be applied.
        (ExpectedCallData, // -> The expected call data to match. Note that the calldata is now held as the key instead of inside this struct.
        u64) // -> The actual amount of times this call was seen
    >
>
````